internal/csi-addons/networkfence/fencing.go

```diff
@@ -212,9 +213,13 @@ func (ac *activeClient) fetchIP() (string, error) {
 	clientInfo := ac.Inst
 	parts := strings.Fields(clientInfo)
 	if len(parts) >= 2 {
-		ip := strings.Split(parts[1], ":")[0]
-
-		return ip, nil
+		// use a regular expression to find an IPv4 or IPv6 address in the string
```

Comment:
you can use net.ParseIP to find the IP address from string https://go.dev/play/p/PSqKE6XxZtO

Reply:
Thanks @Madhu-1 for sharing, updated the code with net.ParseIP and a few additional parsing as the expected format of clientInfo is "inst": "client.4305 172.21.9.34:0/422650892"
PTAL